include/soci/once-temp-type.h

```diff
@@ -10,8 +10,9 @@
 
 #include "soci/ref-counted-statement.h"
 #include "soci/prepare-temp-type.h"
+#include "soci/soci-platform.h"
 
-#if __cplusplus >= 201103L
+#if SOCI_HAVE_NOEXCEPT
```

Comment:
For consistency we should use `#ifdef` for testing `SOCI_HAVE_XXX` (see the existing `HAVE_BOOST` and `SOCI_HAVE_CPP11`.